modules/querying/pages/select-statement/index.adoc

```diff
@@ -0,0 +1,2866 @@
+If you choose to omit the type, you must give the step vertex set an alias.
+
+[.wrap,gsql]
+----
+Result = SELECT tgt
+    FROM :s -(<LIKES)- Person:tgt
+    WHERE tgt.firstName == "Viktor" AND tgt.lastName == "Akhiezer"
+----
+--
+Vertex set::
++
+--
+A source vertex set can also be represented by a vertex set variable.
+[.wrap,gsql]
+----
+CREATE QUERY countFriendsOf2(vertex<person> seed) FOR GRAPH friendNet
+{
+    SumAccum<INT> @@numFriends = 0;
+    seedSet = { seed };
+    friends = SELECT v FROM seedSet:s -((friend | coworker):e)- :v
+      ACCUM @@numFriends +=1;
+    PRINT @@numFriends;
+}
+----
+--
+====
+
+==== Step vertex set
+A vertex set that represents a step in a path pattern.
+Compared with <<_source_vertex_set,source vertex set>>, step vertex sets have more flexibility in how they are denoted.
+A step vertex set can be denoted by one of the following:
+
+* `_` or `ANY`, or omitted.
+If the step vertex type is omitted, you must give the step vertex set an alias.
+* Vertex type
+* A vertex set variable
+* A global accumulator
+
+Optionally, you can give a source vertex set an alias by appending the alias after a colon``:``.
+Although declaring an alias is optional, TigerGraph strongly suggests that you declare them.
+In the later clauses of the `SELECT` block , you can only refer to vertex sets in the `FROM` clause by their aliases.
+
+
+.EBNF for step vertex set
+[.wrap,ebnf]
+----
+stepVertexSet := [stepVertexTypes] [":" vertexAlias]
+stepVertexTypes := atomicVertexType | "(" vertexSetType ["|" vertexSetType]* ")"
+atomicVertexType := "_" | ANY | vertexSetType
+vertexSetType := vertexType | vertexSetVariableName | globalAccumName
+----
+
+Belows are a few examples of valid step vertex sets in `SELECT` statements:
+
+[tabs]
+====
+Vertex type::
++
+--
+[.wrap,gsql]
+----
+Result = SELECT tgt
+    FROM Person:tgt -(<LIKES_REVERSE)- (Comment|Post):src <1>
+    WHERE tgt.firstName == "Viktor" AND tgt.lastName == "Akhiezer"
+    ACCUM CASE WHEN src.type == "Comment" THEN
+        tgt.@commentCnt += 1
+        WHEN src.type == "Post" THEN
+            tgt.@postCnt += 1
+        END;
+----
+<1> `Person` is a vertex type.
+`(Comment | POST)` combines two vertex types.
+--
+Any type::
++
+--
+You can use `_` or `ANY` to represent any vertex types.
+You can also choose to omit the step vertex type altogether to represent any vertex type.
+If you choose to omit the type, you must give the step vertex set an alias.
+
+[.wrap,gsql]
+----
+Result = SELECT s
+    FROM Person:s -(LIKES>)- :tgt
+    WHERE s.firstName == "Viktor" AND s.lastName == "Akhiezer"
+----
+--
+Vertex set::
++
+--
+A step vertex set can also be represented by a vertex set variable.
+[.wrap,gsql]
+----
+CREATE QUERY countFriendsOf2(vertex<person> seed) FOR GRAPH friendNet
+{
+    SumAccum<INT> @@numFriends = 0;
+    seedSet = { seed };
+    friends = SELECT v FROM seedSet:s -((friend | coworker):e)- :v
```

Comment:
In this case the step vertex set is the vertex set associated with the alias `v`, so it would be more appropriate for the FROM clause to be something like:
```
FROM :s -((friend|coworker):e)- seedSet:v
```